libkbfs/tlf_handle_resolve.go

```diff
@@ -34,14 +34,19 @@ type resolvableUser interface {
 	//   - return a non-zero nameIDPair;
 	//   - return a non-zero keybase1.SocialAssertion;
 	//   - return a non-nil error.
-	resolve(context.Context) (nameIDPair, keybase1.SocialAssertion, error)
+	//
+	// The TLF ID may or may not be filled in, depending on whether
+	// the name is backed by an implicit team.
+	resolve(context.Context) (
+		nameIDPair, keybase1.SocialAssertion, tlf.ID, error)
 }
 
 func resolveOneUser(
 	ctx context.Context, user resolvableUser,
 	errCh chan<- error, userInfoResults chan<- nameIDPair,
-	socialAssertionResults chan<- keybase1.SocialAssertion) {
-	userInfo, socialAssertion, err := user.resolve(ctx)
+	socialAssertionResults chan<- keybase1.SocialAssertion,
+	idResults chan<- tlf.ID) {
```

Comment:
I'll change this to make the channel have only a size-1 buffer, and `resolveOneUser` will error if it can't send anything on it.  Then I'll use an explicit `select`, rather than `range`, to get the value from the channel.